### PR TITLE
First iteration of the manage translations form

### PR DESCRIPTION
**Pull request by carolinedanzi**
### Description
1. Adds route endpoints for managing program translations in `routes`
3. Adds `AdminProgramTranslationsController` to handle editing and updating localizations 
2. Adds a form for updating program text translations (includes `ProgramTranslationForm` for binding, and `ProgramTranslationView` for displaying)

An admin can manage translations for a given program by:
1. Create a new draft program
2. Click "Manage Translations" on the program index view
3. Select the language they want to update
4. Add display name and description for that language
5. Submit

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)
#777 

Screen Shot 2021-04-22 at 2 28 44 PM
Screen Shot 2021-04-22 at 6 53 47 PM

### CODE REVIEW

File: universal-application-tool-0.0.1/app/controllers/admin/AdminProgramTranslationsController.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    String displayName = translations.getDisplayName();
+    String displayDescription = translations.getDisplayDescription();
+
+    try {
+      ErrorAnd<ProgramDefinition, CiviFormError> result =
+          service.updateLocalization(id, updatedLocale, displayName, displayDescription);
+      if (result.isError()) {
+        String errorMessage = joinErrors(result.getErrors());
+        return ok(
+            translationView.render(
```
Comment (hmhsu) [inline universal-application-tool-0.0.1/app/controllers/admin/AdminProgramTranslationsController.java:89]: praise: thanks for handling the errors!

File: universal-application-tool-0.0.1/app/views/admin/programs/ProgramIndexView.java [hunk 2]
```diff
@@ -185,6 +187,24 @@ Tag maybeRenderEditLink(
...
 
+  private Tag maybeRenderManageTranslationsLink(Optional<ProgramDefinition> draftProgram) {
+    if (draftProgram.isPresent()) {
+      String linkText = "Manage Translations →";
+      String linkDestination =
+          routes.AdminProgramTranslationsController.edit(
+                  draftProgram.get().id(), LocalizationUtils.DEFAULT_LOCALE.toLanguageTag())
+              .url();
+      return new LinkElement()
+          .setId("program-edit-link-" + draftProgram.get().id())
```
Comment (hmhsu) [inline universal-application-tool-0.0.1/app/views/admin/programs/ProgramIndexView.java:198]: Suggestion (blocking): would this conflict with program edit link? Would you use a different id?

**Review by hmhsu — APPROVED** [review]
The translation view looks really nice. One non-blocking question: do we now have two ways to update the default locale: one in program edit page and one in the translation page?

File: universal-application-tool-0.0.1/app/views/admin/programs/ProgramIndexView.java (hunk 2 above)
Comment (carolinedanzi) [reply universal-application-tool-0.0.1/app/views/admin/programs/ProgramIndexView.java:198]: Ah great catch! Copy/paste error - I updated to `program-translations-link-%id%`

**Comment by carolinedanzi** [comment]
> The translation view looks really nice. One non-blocking question: do we now have two ways to update the default locale: one in program edit page and one in the translation page?

Great question! Yes we do have two ways, that's a great point. I made issue #875 to capture this side effect

**Pull request merged**